### PR TITLE
 	Remove unneeded generated end extra line and add extra empty lines between trait function declarations

### LINKED ISSUE #188: Extra empty line at the end of struct
When generating a `struct`, it ends like this:

```Rust
struct Foo {
    fn a() {}

}
```

I'm working on fixing this.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #188.

### CODE REVIEW

File: src/codegen/function.rs [hunk 1]
```diff
@@ -14,10 +14,10 @@ use writer::ToCode;
 
 pub fn generate(w: &mut Write, env: &Env, analysis: &analysis::functions::Info,
     in_trait: bool, only_declaration: bool, indent: usize) -> Result<()> {
-
     let mut commented = false;
     let mut comment_prefix = "";
     let mut pub_prefix = if in_trait { "" } else { "pub " };
+    try!(writeln!(w, ""));
```
Comment (gkoz) [inline src/codegen/function.rs]: Without an `if` this has an additional effect in trait definitions. Is that inteded? It doesn't match the commit/PR note.

**Comment by GuillaumeGomez** [comment]
Commit message has been updated.

**Comment by gkoz** [comment]
Okay, considering we'll be putting the docs there, adding blank lines between trait function items is a good thing.
Thanks!
@homu r+

**Comment by gkoz** [comment]
Something's wrong with homu I guess.

**Pull request merged**